src/editor/lib/blocks/ucd-theme-social-media/edit.js

```diff
@@ -0,0 +1,143 @@
+    setAttributes({twitterUrl: 'https://twitter.com/UCDavisLibrary'});
+    setAttributes({instagramUrl: 'https://www.instagram.com/ucdavislibrary/'});
+    setAttributes({youtubeUrl: 'https://www.youtube.com/channel/UCRjjo_jpHml_Z3_5ctYq1lA'});
+  }, []);
+
+  const startingModalData = {
+    facebookUrl: !attributes.facebookUrl ? '' : 'https://www.facebook.com/UCDavisLibrary/',
+    twitterUrl: !attributes.twitterUrl ? '' : 'https://twitter.com/UCDavisLibrary',
+    instagramUrl: !attributes.instagramUrl ? '' : 'https://www.instagram.com/ucdavislibrary/',
+    youtubeUrl: !attributes.youtubeUrl ? '' : 'https://www.youtube.com/channel/UCRjjo_jpHml_Z3_5ctYq1lA',
+    linkedinUrl: ''
+  };
+  const [ modalIsOpen, setModalOpen ] = useState( false );
+  const [ modalMode, setModalMode ] = useState( 'Add' );
+  const [ modalData, setModalData ] = useState( startingModalData );
+
+  const closeModal = () => {
+    setModalOpen(false);
+  }
+
+  const onAddLinks = () => {
+    setModalMode('Add');
+    setModalData(startingModalData);
+    setModalOpen(true);
+  }
+
+  const onModalSave = () => {
+    setModalOpen(false);
+  }
+
+  const onFacebookUrlChange = (facebookUrl) => {
+    const data = {...modalData, facebookUrl};
+    setModalData(data);
+    setAttributes({facebookUrl});
+  }
+
+  const onTwitterUrlChange = (twitterUrl) => {
+    const data = {...modalData, twitterUrl};
+    setModalData(data);
+    setAttributes({twitterUrl});
+  }
+
+  const onInstagramUrlChange = (instagramUrl) => {
+    const data = {...modalData, instagramUrl};
+    setModalData(data);
+    setAttributes({instagramUrl});
+  }
+
+  const onYoutubeUrlChange = (youtubeUrl) => {
+    const data = {...modalData, youtubeUrl};
+    setModalData(data);
+    setAttributes({youtubeUrl});
+  }
+
+  const onLinkedinUrlChange = (linkedinUrl) => {
+    const data = {...modalData, linkedinUrl};
+    setModalData(data);
+    setAttributes({linkedinUrl});
+  }
+
+  return html`
+    <div ...${ blockProps }>
+      <${BlockControls} group="block">
+        <${ToolbarButton} 
+            icon=${UCDIcons.render('link')} 
+            onClick=${() => onAddLinks()} 
+            isPressed=${modalIsOpen}
+            label="Add Social Media Links"
+        />
+      </${BlockControls}>
+      ${!attributes.facebookUrl && !attributes.twitterUrl && !attributes.instagramUrl && !attributes.youtubeUrl && !attributes.linkedinUrl && html`
+        <div>
+          <div style=${{backgroundColor: '#1da1f2', width: '40px', height: '40px', display: 'inline-block'}}></div>
+          <div style=${{backgroundColor: '#1778f2', width: '40px', height: '40px', display: 'inline-block'}}></div>
+          <div style=${{backgroundColor: '#f00075', width: '40px', height: '40px', display: 'inline-block'}}></div>
+        </div>
+      `}
+      ${attributes.facebookUrl ? html`
+        <ucdlib-icon icon='ucd-public:facebook' class="media-icon"></ucdlib-icon>
```

Comment:
sorry, i misunderstood your question about using the icons.
don't use `ucdlib-icon`. try to follow the brand styles as closely as possible: 
 - host the [sprite sheet on the site](https://github.com/UCDavisLibrary/ucdlib-theme-wp/tree/sandbox/assets/img). 
 - Then add a css rule that modifies `.social-follow a` to be `background-image:url(/wp-content/themes/ucdlib-theme-wp/assets/img/social-follow.svg)`

does that make sense?

Reply:
I think so, but the spritesheet is an svg with all 5 icons. Is there a way to set the background-image to a specific viewport of an svg? 
Cosmetically in the UI they match the brand styles very closely, just technically they're wrapped in `ucdlib-icon` markup.

Reply:
yeah, the patternlab styles should do it for you if you just set the class: 
<img width="404" alt="image" src="https://user-images.githubusercontent.com/34284106/173683575-541f49b8-da7c-40d8-bd0d-6e6ef15973b8.png">

http://dev.webstyleguide.ucdavis.edu/redesign/?p=molecules-social-follow